### ISSUE TITLE
Cell probability question

### ISSUE THREAD

**Issue by PalatinTech** [broadinstitute/CellBender#134]
Dear Authors
I have recently started using cell bender for my snRNA-seq QC analysis. I have snRNA-seq data from 10x genomics platform with cellranger output for each sample. We targeted 5000 nuclei during isolation but recovered very few nuceli with sufficent RNA content. According to cellranger there are 2983 non-empty droplet and only 1059 have UMI counts> 200. I ran cellbender on cellranger raw_feature_bc_matrix.h5 file and got 7000 non-empty droplets and 1436 droplets with UMI counts>200. Cell bender is calling way more  number of non-empty droplets and as per my understanding these droplets contain debris. 
Here's cell ranger barcode rank plot

![SC1](https://user-images.githubusercontent.com/105299970/174831982-13d9467e-56c7-4453-a246-02557f27d82e.jpg)

Cellbender input:
Inputfile: raw_feature_bc_matrix.h5
--expected-cells 500
--total-droplets-included 7000
--fpr 0.01
--epochs 150

Cellbender output:
cellbender:remove-background: Running remove-background
cellbender:remove-background: Loading data from file raw_feature_bc_matrix.h5
cellbender:remove-background: CellRanger v3 format
cellbender:remove-background: Trimming dataset for inference.
cellbender:remove-background: Including 19377 genes that have nonzero counts.
cellbender:remove-background: Prior on counts in empty droplets is 72
cellbender:remove-background: Prior on counts for cells is 3177
cellbender:remove-background: Excluding barcodes with counts below 36
cellbender:remove-background: Using 500 probable cell barcodes, plus an additional 6500 barcodes, and 70302 empty droplets.
cellbender:remove-background: Largest surely-empty droplet has 296 UMI counts.
![SC4](https://user-images.githubusercontent.com/105299970/174832758-5c9034a0-2acc-4ee8-9d44-0857ceb95ddb.jpg)
![SC3](https://user-images.githubusercontent.com/105299970/174832763-1784dd88-2835-449b-8723-34d6c352a7c3.jpg)
![SC2](https://user-images.githubusercontent.com/105299970/174832764-9116ca26-b924-494b-866d-ca5388b4ba17.jpg)

I would like to know why cellbender is calling most of the empty droplets as non-empty. I would like to peform clustering and gene expression analysis using seurat and low viable cell count will impact downstream analysis. Can you please suggest how to handle this data using CellBender? Do i need to adjust the input parameters? 
Greatly appreciate your response.
Thanks
Priyanka

**Comment by sjfleming** [comment]
Hi Priyanka,

I do see that CellBender is probably calling too many cells in your dataset.  I think the issue here has to do with the fact that the "empty droplet plateau" in your dataset has much more dispersion in terms of UMI counts than we typically see in a 10x experiment.  For example, something more like this is more typical, where the empty drops have a tighter distribution of UMI counts (https://cf.10xgenomics.com/samples/cell-exp/3.0.0/heart_10k_v3/heart_10k_v3_web_summary.html).
![image](https://user-images.githubusercontent.com/10214815/175322272-6482c091-e380-4edf-b399-34416ff600b2.png)

Unfortunately, I think what happens in cases like yours is that CellBender struggles to identify cells because the distributional assumption it makes about empty droplets (that their counts ultimately come from a Poisson distribution) does not seem to hold very well.  Since our model cannot explain the counts in those empties using a Poisson, it must say they are cells, and then it is allowed to model them with a more flexible negative binomial distribution.

In terms of "why is it happening, for this experiment?"... that is a tough question.  I've seen experiments like this before, where it's very hard, by eyeball, to see what's a cell and what's not.  In some cases, this has been due to some kind of fundamental failure in the library prep.  (Especially when it's so hard to tell what's a cell by eye on the CellBender "determination of which barcodes contain cells" plot, where the black line really shows no elbow at all...)  In some of these cases, it's a snRNA-seq prep where there was a lot of cytoplasmic debris / material that could not easily be cleaned up very well.  And in those cases, we saw a UMI curve more like what you see.  CellBender calls everything "non-empty" because that's all it can do to explain the data... which is kind of unfortunate, but sadly I do not know of a way to get around it.

That being said... from a practical standpoint, when you use "--total-droplets-included" 7000, you are explicitly telling CellBender that all droplets after that point are "surely empty", and so CellBender uses that information when it's learning background noise.  And so CellBender, even though it's not calling cells in the way we think it should, should still be removing noise counts as expected (or as best as it can).  What I would try in your case is to use a  "--total-droplets-included" number that is on the small side... so don't include much of what you're pretty sure is empty.  Then I would go ahead and use the cleaned up CellBender data, but either use a different method to call cells, or just use all the CellBender called cells but QC them afterward.  You mentioned that a 200 UMI cutoff gets you down to about 1400 cells.  I would recommend using these (and maybe doing some additional cell QC) downstream.
